packages/typeslayer/src-tauri/src/commands/actions.rs

```diff
@@ -86,19 +83,14 @@ fn capture_window_to_png<R: tauri::Runtime>(window: &WebviewWindow<R>) -> Result
 }
 
 #[tauri::command]
-pub async fn open_file(state: State<'_, Arc<Mutex<AppData>>>, path: String) -> Result<(), String> {
+pub async fn open_file(state: State<'_, &Mutex<AppData>>, path: String) -> Result<(), String> {
+    let data = state.lock().await;
```

Comment:
Notice how I can just lock `state` once and call it good? This was my main motivation for switching over since being careful about exactly how you hold the lock is just annoying.